cli/src/cmd/run.js

```diff
@@ -207,13 +208,21 @@ export async function runInDebug({
   spinner.start('Observing task...');
   const taskId = await iexec.deal.computeTaskId(dealid, 0);
   const taskObservable = await iexec.task.obsTask(taskId, { dealid: dealid });
+  const taskTimeoutWarning = setTimeout(() => {
+    spinner.warn('Task is taking longer than expected...');
+    spinner.start('Observing task...');
+  }, TASK_OBSERVATION_TIMEOUT);
   await new Promise((resolve, reject) => {
     taskObservable.subscribe({
       next: () => {},
       error: (e) => {
         reject(e);
+        clearTimeout(taskTimeoutWarning);
+      },
+      complete: () => {
+        resolve(undefined);
+        clearTimeout(taskTimeoutWarning);
       },
-      complete: () => resolve(undefined),
     });
```

Comment:
use `await new Promise(/*...*/).finally(() => clearTimeout(taskTimeoutWarning))` to handle both resolve and reject in the same place